app/controllers/result_filters/provider_controller.rb

```diff
@@ -18,7 +18,9 @@ def new
         flash[:error] = [I18n.t("location_filter.fields.provider")]
         redirect_back
       elsif @provider_suggestions.count == 1
-        redirect_to results_path(filter_params.merge(query: @provider_suggestions.first.provider_name))
+        redirect_to results_path(
+          remove_previous_parameters(filter_params.merge(query: @provider_suggestions.first.provider_name)),
```

Comment:
Could also be elligible for `filter_params_without_previous` (or similar), as mentioned below in the view.